exo-add/src/cli.ls

```diff
@@ -60,6 +62,15 @@ function help
   console.log help-message
 
 
+function check-for-service {service-role, services}
+  existing-services = []
+  for protection-level of services
+    if services[protection-level]
+      existing-services.push Object.keys that
+  if flatten existing-services |> (.includes service-role)
```

Comment:
I'm not opposed to using flatten here, but don't like that there is so much low-level logic in this high-level method. Ideally this method calls a helper with the data that the helper needs, and the helper massages it.